src/UnderlineNav2/UnderlineNav.test.tsx

```diff
@@ -119,3 +146,55 @@ describe('UnderlineNav', () => {
     expect(heading.textContent).toBe('Repository navigation')
   })
 })
+
+describe('Keyboard Navigation', () => {
+  it('should move focus to the next/previous item on the list with horizontal arrow keys', () => {
+    const {getByText} = render(<ResponsiveUnderlineNav />)
+    const item = getByText('Code').closest('a') as HTMLAnchorElement
+    const nextItem = getByText('Issues').closest('a') as HTMLAnchorElement
+    // Focus first item
+    item.focus()
+    // Press right arrow
+    fireEvent.keyDown(item, arrowRight)
+    // focus should be on the next item
+    expect(nextItem).toHaveFocus()
+    expect(nextItem.getAttribute('tabindex')).toBe('0')
+    fireEvent.keyDown(nextItem, arrowLeft)
+    // focus should be on the previous item
+    expect(item).toHaveFocus()
+    expect(nextItem.getAttribute('tabindex')).toBe('-1')
+    expect(item.getAttribute('tabindex')).toBe('0')
+  })
+  it('should move focus to the next/previous item on the list with vertical arrow keys', () => {
+    const {getByText} = render(<ResponsiveUnderlineNav />)
+    const item = getByText('Code').closest('a') as HTMLAnchorElement
+    const nextItem = getByText('Issues').closest('a') as HTMLAnchorElement
+    // Focus first item
+    item.focus()
+    // Press down arrow
+    fireEvent.keyDown(item, arrowDown)
+    // focus should be on the next item
+    expect(nextItem).toHaveFocus()
+    expect(nextItem.getAttribute('tabindex')).toBe('0')
+    // Press up arrow
+    fireEvent.keyDown(nextItem, arrowUp)
+    // focus should be on the previous item
+    expect(item).toHaveFocus()
+    expect(nextItem.getAttribute('tabindex')).toBe('-1')
+    expect(item.getAttribute('tabindex')).toBe('0')
+  })
+  it('should move focus to the next/previous item on the list with the tab key', () => {
+    const {getByText} = render(<ResponsiveUnderlineNav />)
+    const item = getByText('Code').closest('a') as HTMLAnchorElement
```

Comment:
When do you think it would be better to do `getByText('Code').closest('a')` versus `getByRole('link', { name: 'Code' })`

Reply:
Hmm 🤔  I would use them interchangeable but probably not the best? I can't think of any case right now where using the element.closest('a') will be the best.  Do you have an example? That said, `getByRole` is more declarative so I'll opt in using this, thank for pointing this out :)